README.md

```diff
@@ -6,34 +6,81 @@
 
 At the core of Dagger.jl is a scheduler heavily inspired by [Dask](http://dask.pydata.org/en/latest/). It can run computations represented as [directed-acyclic-graphs](https://en.wikipedia.org/wiki/Directed_acyclic_graph) (DAGs) efficiently on many Julia worker processes.
 
-# DAG creation interface
 
-Here is an example DAG:
+## Installation
+
+You can install it by typing
+
+```julia
+julia> ] add Dagger
+```
+
+## Usage
+
+Once installed, the `Dagger` package can by used by typing
+
+```julia
+using Dagger
+```
+
+The main function for using Dagger is *delayed*
+
+`delayed(f; options...)`
+
+When called creates a `Thunk` object representing a call to function `f` with the given arguments. If it is called with other thunks as input, then they form a graph with input nodes directed at the output. The function `f` get the result of the input Thunks.
+
+Here is a very simple example DAG:
 
 ```julia
 using Dagger
 
-p = delayed(f; options...)(42)
-q = delayed(g)(p)
-r = delayed(h)(53)
+add1(value) = value+1
+add2(value) = value+2
+combine(a...) = sum(a)
+
+p = delayed(add1)(4)
+q = delayed(add2)(p)
+r = delayed(add1)(3)
 s = delayed(combine)(p, q, r)
+
+collect(s)
```

Comment:
For clarity, maybe make this `@assert collect(s) == 16` so that it's clear that `collect` returns the final result (and also illustrates how `delayed` is just a set of "regular" function calls).